src/Frontend-GNOME/Views/MessageTextView.cs

```diff
@@ -406,6 +407,25 @@ public void AddMessage(MessageModel msg, bool addLinebreak, bool showTimestamps)
                     } else {
                         buffer.Insert(ref iter, fmsgti.Text);
                     }
+                } else if (msgPart is ImageMessagePartModel) {
+                    var imgpart = (ImageMessagePartModel) msgPart;
+                    try {
+                        var uri = new Uri(imgpart.ImageFileName);
+                        if (uri.Scheme == "smuxi-emoji") {
+                            var emojiPath = System.IO.Path.Combine("/tmp/emoji/assets/images", uri.Host + ".png");
+                            var emojiFile = new FileInfo(emojiPath);
+                            if (emojiFile.Exists && emojiFile.Length > 0) {
+                                var pix = new Gdk.Pixbuf(emojiPath);
+                                buffer.InsertPixbuf(ref iter, pix);
+                                continue;
```

Comment:
I would prefer if the regular expected case is not the nested one.. continue in success code code should be avoided